lab_3/main.py

```diff
@@ -59,49 +109,95 @@ def encode_corpus(storage: LetterStorage, corpus: tuple) -> tuple:
     :param corpus: a tuple of sentences
     :return: a tuple of the encoded sentences
     """
-    pass
+    if not isinstance(corpus, tuple) or type(storage) in [int, str, list, tuple, dict, float, bool] or \
```

Comment:
`type(storage) in [int, str, list, tuple, dict, float, bool]` - way too specific.